stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/textinput/StreamMessageInputView.kt

```diff
@@ -0,0 +1,165 @@
+
+            typedArray.getDrawable(R.styleable.StreamMessageInputView_attachmentButtonIcon)
+                ?.let(this::setImageDrawable)
+
+            DrawableCompat.setTintList(
+                drawable,
+                getColorList(
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_attachmentButtonIconColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_dark)
+                    ),
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_attachmentButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_white)
+                    ),
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_attachmentButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_light)
+                    )
+                )
+            )
+
+            layoutParams.width = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_attachmentButtonWidth,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_width)
+            )
+
+            layoutParams.height = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_attachmentButtonHeight,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_height)
+            )
+        }
+    }
+
+    private fun configLightningButton(typedArray: TypedArray) {
+        binding.ivOpenEmojis.run {
+            isVisible =
+                typedArray.getBoolean(R.styleable.StreamMessageInputView_showLightningButton, true)
+
+            typedArray.getDrawable(R.styleable.StreamMessageInputView_lightningButtonIcon)
+                ?.let(this::setImageDrawable)
+
+            DrawableCompat.setTintList(
+                drawable,
+                getColorList(
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_dark)
+                    ),
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_white)
+                    ),
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_light)
+                    )
+                )
+            )
+
+            layoutParams.width = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_lightningButtonWidth,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_width)
```

Comment:
Sometimes I start thinking that we need yet another module, something like `common-utils` because we seem to have extension function for stuff like that (`context.resources.getDimensionPixelSize`) everywhere :)

Reply:
These things should go into `stream-chat-android` for now, and be marked with `@InternalStreamChatApi`.

Later on, we can pull out most of the non-UI parts of that module into a `stream-chat-android-ui-core` (name pending), and have both UI implementations depend on _that_ module.

Reply:
Not sure if I get it. `getDimensionPixelSize` is from `android.content.res`. It is not our code

Reply:
I think he meant [this extension](https://github.com/GetStream/stream-chat-android/blob/1f16150c4025e5dc14ab7203f39d600998e92b55/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/common/Extensions.kt#L32-L34) specifically, and that we should have it available everywhere in the different modules, because we make these kinds of calls a lot. For now, it's fine to do it manually, we can revisit extracting this to simplify later.

Reply:
Oh I see.